### PR TITLE
[onert] Get CompilerOptions out of Compiler

### PR DESCRIPTION
**Pull request by glistening**
CompilerOptions outlives Compiler.
CompilerOptions are prepared before Compiler::compile()
  - nnfw_set_available_backecnds(), nnfw_set/get_config().

Signed-off-by: Sanggyu Lee <sg5.lee@samsung.com>

Related: https://github.com/Samsung/ONE/pull/9271#issuecomment-1158387870

### CODE REVIEW

**Review by chunseoklee — APPROVED** [review]

File: runtime/onert/core/include/compiler/Compiler.h [hunk 1]
```diff
@@ -133,7 +142,7 @@ class Compiler
   // have to add allocate non-constant tensor memory of executors in execution time when each
   // subgraph is called.
   State _state;
-  CompilerOptions _options;
+  CompilerOptions &_options;
```
Comment (ragmani) [inline runtime/onert/core/include/compiler/Compiler.h:145]: Why did you change this to a reference member?

Comment (glistening) [reply runtime/onert/core/include/compiler/Compiler.h:145]: @ragmani `nnfw_set_available_backends()`, `nnfw_set_config()` and `nnfw_get_config()` reads and modifies the compiler option of `nnfw_session`. `CompilerOptions` should be a reference to `nnfw_session`'s `CompilerOptions`.

Comment (ragmani) [reply runtime/onert/core/include/compiler/Compiler.h:145]: How about change it a pointer member instead?

Comment (glistening) [reply runtime/onert/core/include/compiler/Compiler.h:145]: I used reference because
1) I don't want null check which is necessary for pointer.
2) Using pointer type requires more changes.

Why do you prefer pointer?

Comment (ragmani) [reply runtime/onert/core/include/compiler/Compiler.h:145]: It is nothing. Ignore it.

**Review by ragmani — APPROVED** [review]
